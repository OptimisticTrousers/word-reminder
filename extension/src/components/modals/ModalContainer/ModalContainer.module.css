.modal {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: var(--modal-bkg-color);
  z-index: 5;
}

.modal__container {
  border-radius: 8px;
  box-shadow: 4px 4px 20px rgb(0 0 0 / 30%);
  display: flex;
  flex-direction: column;
  min-height: min(100%, 428px);
  min-width: 310px;
  position: relative;
  width: min(100%, 500px);
}

.modal__header {
  margin-bottom: 0.5rem;
}

.modal__header button.modal__button {
  position: absolute;
  border-radius: 50%;
  background-color: transparent;
  right: 12px;
  top: 12px;
  padding: 0;
  color: var(--secondary-icon-color);
}

.modal__header button.modal__button:focus {
  outline: auto;
  box-shadow: none;
}

.modal__header .modal__button:hover {
  color: var(--tertiary-icon-color);
  transform: scale(1.075);
  background-color: transparent;
}

.modal__icon {
  height: 32px;
  width: 32px;
}

.modal__title-bar {
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 60px;
  border-bottom: 1px solid var(--break-color);
}

.modal__title {
  color: var(--text-color);
}
