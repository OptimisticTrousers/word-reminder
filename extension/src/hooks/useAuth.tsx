// import { useMutation } from "@tanstack/react-query";
// import { useEffect, useState } from "react";
// import { useNavigate } from "react-router-dom";
// import { toast } from "react-toastify";

// const useAuth = () => {
//     const navigate = useNavigate();
//     // const [isAuthenticated, setIsAuthenticated] = useState(false);

//     // const {data, status, error, mutuate} = useMutation({mutationFn: () => {
//     //     return fetch(`${import.meta.env.VITE_API_DOMAIN}/current`)
//     // }})

//     const { data: loginData, status: loginStatus, error: loginError, mutate: loginMutate } = useMutation({
//         mutationFn: (formData) => {
//             return fetch(`${import.meta.env.VITE_API_DOMAIN}/login`, formData)
//         }, onSuccess: () => {
//             toast.success("You have successfully logged in!");
//             navigate("/")
//         }, onError: () => {
//             toast.error("There was an issue logging in!")
//         }
//     })

//     const { data: registerData, status: registerStatus, error: registerError, mutate: registerMutate } = useMutation({
//         mutationFn: (formData) => {
//             return fetch(`${import.meta.env.VITE_API_DOMAIN}/register`, formData)
//         }
//         , onSuccess: () => {
//             toast.success("You have successfully registered!");
//             navigate("/")
//         }, onError: () => {
//             toast.error("There was an issue registering!")
//         }
//     })

//     const { data: logoutData, status: logoutStatus, error: logoutError, mutate: logoutMutate } = useMutation({
//         mutationFn: (formData) => {
//             return fetch(`${import.meta.env.VITE_API_DOMAIN}/logout`, formData)
//         }
//         , onSuccess: () => {
//             toast.success("You have successfully logged out!");
//             navigate("/")
//         }, onError: () => {
//             toast.error("There was an issue logging out!")
//         }
//     })


//     useEffect(() => {

//     }, [])

//     return {
//         login: {
//             data: loginData,
//             status: loginStatus,
//             error: loginError,
//             mutate: loginMutate
//         }, register: {
//             data: registerData,
//             status: registerStatus,
//             error: registerError,
//             mutate: registerMutate
//         }, logout: {
//             data: logoutData, status: logoutStatus, error: logoutError, mutuate: logoutMutate
//         }
//     };
// }

// export default useAuth;