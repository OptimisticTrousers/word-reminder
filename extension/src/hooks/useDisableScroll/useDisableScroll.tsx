import { useEffect } from "react";

export const useDisableScroll = () => {
  useEffect(() => {
    document.documentElement.style.overflow = "hidden";
    return () => {
      document.documentElement.style.removeProperty("overflow");
    };
  }, []);
};
