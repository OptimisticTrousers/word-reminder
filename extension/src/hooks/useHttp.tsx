type HttpMethods = "GET" | "POST" | "PUT" | "DELETE";

const useHttp = () => {
  const request = async (
    url: string,
    method: HttpMethods,
    options: RequestInit,
    body?: unknown
  ) => {
    const response = await fetch(url, {
      method,
      mode: "cors",
      credentials: "include",
      headers: {
        "Content-Type": "application/json",
      },
      ...options,
      body: JSON.stringify(body),
    });

    const json = await response.json();

    return json;
  };

  const get = async (url: string, options: RequestInit = {}) => {
    return request(url, "GET", options);
  };

  const post = async (
    url: string,
    body: unknown,
    options: RequestInit = {}
  ) => {
    return request(url, "POST", options, body);
  };

  const put = async (url: string, body: unknown, options: RequestInit = {}) => {
    return request(url, "PUT", options, body);
  };

  const remove = async (url: string, options: RequestInit = {}) => {
    return request(url, "DELETE", options);
  };

  return { get, post, put, remove };
};

export default useHttp;
