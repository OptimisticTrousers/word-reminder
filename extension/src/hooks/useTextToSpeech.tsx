import { useState, useEffect } from "react";

const useTextToSpeech = (text: string) => {
  const [utterance, setUtterance] = useState<SpeechSynthesisUtterance | null>(
    null
  );

  useEffect(() => {
    const synth = window.speechSynthesis;
    const u = new SpeechSynthesisUtterance(text);

    setUtterance(u);

    return () => {
      synth.cancel();
    };
  }, [text]);

  const handlePlay = () => {
    const synth = window.speechSynthesis;

    synth.speak(utterance!);
  };

  return { handlePlay };
};

export default useTextToSpeech;
