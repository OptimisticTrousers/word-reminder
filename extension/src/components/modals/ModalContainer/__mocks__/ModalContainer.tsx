import { Props } from "../ModalContainer";

export function ModalContainer({ title, toggleModal, children }: Props) {
  return (
    <div>
      <h2>{title}</h2>
      <button aria-label="Close modal" onClick={toggleModal}>
        X
      </button>
      <div>{children}</div>
    </div>
  );
}
