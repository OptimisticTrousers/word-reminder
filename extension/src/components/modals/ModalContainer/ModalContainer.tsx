import { SquareX } from "lucide-react";
import { ReactNode } from "react";
import CSSModules from "react-css-modules";

import { useDisableScroll } from "../../../hooks/useDisableScroll";
import styles from "./ModalContainer.module.css";

interface Props {
  children: ReactNode;
  title: string;
  toggleModal: () => void;
}

export const ModalContainer = CSSModules(
  function ({ children, title, toggleModal }: Props) {
    useDisableScroll();

    function handleClose() {
      toggleModal();
    }

    return (
      <div
        styleName="modal"
        aria-modal="true"
        role="dialog"
        aria-labelledby="modal-title"
      >
        <div styleName="modal__container" className="scale-down">
          <header styleName="modal__header">
            <button
              styleName="modal__button"
              onClick={handleClose}
              aria-label="Close modal"
            >
              <SquareX styleName="modal__icon" />
            </button>
            <div styleName="modal__title-bar" id="modal-title">
              <h2 styleName="modal__title">{title}</h2>
            </div>
          </header>
          {children}
        </div>
      </div>
    );
  },
  styles,
  { allowMultiple: false, handleNotFoundStyleName: "log" }
);
