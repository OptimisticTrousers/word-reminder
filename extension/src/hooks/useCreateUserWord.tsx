import { useMutation } from "@tanstack/react-query";
import useHttp from "./useHttp";

const useCreateUserWord = () => {
  

  return {
    createdUserWord,
    createdUserWordStatus,
    createdUserWordError,
    createdUserWordMutate,
  };
};

export default useCreateUserWord;
