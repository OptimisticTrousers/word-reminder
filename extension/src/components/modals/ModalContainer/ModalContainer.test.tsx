import { render, screen } from "@testing-library/react";
import { ModalContainer } from "./ModalContainer";
import userEvent from "@testing-library/user-event";

import * as hooks from "../../../hooks/useDisableScroll";

describe("ModalContainer", () => {
  it("renders modal with text", () => {
    const mockToggleModal = vi.fn();
    const props = {
      title: "Title",
      toggleModal: mockToggleModal,
    };
    const mockUseDisableScroll = vi.spyOn(hooks, "useDisableScroll");

    const { asFragment } = render(
      <ModalContainer {...props}>
        <div></div>
      </ModalContainer>
    );

    const notification = screen.getByRole("dialog", { name: props.title });
    expect(notification).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
    expect(mockUseDisableScroll).toHaveBeenCalledTimes(1);
    expect(mockUseDisableScroll).toHaveBeenCalledWith();
  });

  it("calls toggleModal function when button is clicked", async () => {
    const mockToggleModal = vi.fn();
    const props = {
      title: "Title",
      toggleModal: mockToggleModal,
    };
    render(
      <ModalContainer {...props}>
        <div></div>
      </ModalContainer>
    );
    const user = userEvent.setup();

    const button = screen.getByRole("button", { name: "Close modal" });
    await user.click(button);

    expect(props.toggleModal).toHaveBeenCalledTimes(1);
    expect(props.toggleModal).toHaveBeenCalledWith();
  });
});
