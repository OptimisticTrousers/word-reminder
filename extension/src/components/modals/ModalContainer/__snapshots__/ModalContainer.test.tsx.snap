// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModalContainer > renders modal with text 1`] = `[Function]`;
